test/graphql/attachments_query_test.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class AttachmentsQueryTest < ActiveSupport::TestCase
+  ATTACHMENTS_QUERY = <<~GRAPHQL
+    query($first: Int) {
+      samples(first: $first) {
+        nodes {
+          id
+          attachments {
+            edges {
+              node {
+                id,
+                metadata,
+                filename,
+                byteSize
+              }
+            }
+          }
+        }
+      }
+    }
+  GRAPHQL
+
+  ATTACHMENTS_URL_QUERY = <<~GRAPHQL
+    query($first: Int) {
+      samples(first: $first) {
+        nodes {
+          id
+          attachments {
+            edges {
+              node {
+                id,
+                filename,
+                attachmentUrl
+              }
+            }
+          }
+        }
+      }
+    }
+  GRAPHQL
+
+  def setup
+    @user = users(:john_doe)
+    @sample = samples(:sample1)
+  end
+
+  test 'attachment query should work' do
+    result = IridaSchema.execute(ATTACHMENTS_QUERY, context: { current_user: @user },
+                                                    variables: { first: 1 })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['samples']
+
+    assert_not_empty data, 'samples type should work'
+    assert_not_empty data['nodes']
+    assert_equal 1, data['nodes'].count
+
+    assert_not_empty data['nodes'][0]['attachments']
+    assert_not_empty data['nodes'][0]['attachments']['edges']
+
+    attachments = data['nodes'][0]['attachments']['edges']
+    assert_equal 2, attachments.count
+
+    assert_equal 'test_file.fastq', attachments[0]['node']['filename']
+    assert_equal 'test_file_A.fastq', attachments[1]['node']['filename']
+
+    assert_equal 2102, attachments[0]['node']['byteSize']
+    assert_equal 2101, attachments[1]['node']['byteSize']
+
+    assert_equal 'fastq', attachments[0]['node']['metadata']['format']
+    assert_equal 'none', attachments[0]['node']['metadata']['compression']
+    assert_equal 'fastq', attachments[1]['node']['metadata']['format']
+    assert_equal 'none', attachments[1]['node']['metadata']['compression']
+  end
+
```

Comment:
Can you also add a test for a sample with paired-end reads

Reply:
Added in https://github.com/phac-nml/irida-next/pull/408/commits/a9e7880fdfd0d774e15793238984a29ab352d2cd